filebeat/prospector/prospector_log.go

```diff
@@ -60,6 +61,19 @@ func (p *ProspectorLog) Init(states []file.State) error {
 func (p *ProspectorLog) Run() {
 	logp.Debug("prospector", "Start next scan")
 
+	// TailFiles is like ignore_older = 1ns and only on startup
+	if p.config.TailFiles {
+		ignoreOlder := p.config.IgnoreOlder
+
+		// Overwrite ignore_older for the first scan
+		p.config.IgnoreOlder = 1
+		defer func() {
+			// Reset ignore_older after first run
+			p.config.IgnoreOlder = ignoreOlder
+			// Disable tail_files after the first run
+			p.config.TailFiles = false
+		}()
+	}
```

Comment:
Should also work. The advantage of the first version is, that in case TailFiles is also used somewhere inside the first scan (which is not the case), it would still work.